RELEASING.md

```diff
@@ -40,3 +40,17 @@
 1. Create a new GitHub release for that tag.  Use "Generate release notes" (against the tag for the previous release)
    as a starting point for the release description.  Then add the "headlines" produced earlier at the top.
 2. Update dependent crates (eg, hyper-rustls, rustls-native-certs, etc.) if this was a semver-incompatible release.
+
+## Maintenance point releases
+
+When point releases for bug fixes and small backwards compatible changes, but `main` contains unreleased breaking 
+changes we follow a modified release process using a longer-lived maintenance branch.
+
+1. Check if there is an existing release branch, e.g. `rel-0.21` for point releases in the `0.21.x` series.
+   - If there is, use that branch.
+   - If there is not, create a new branch from the tag for the previous release, e.g. `git checkout -b rel-0.21 v/0.21.0`.
+2. Make pull-requests for any changes you want to include in the point release, targeted against the release branch.
+   - When possible, we prefer to land compatible changes on the release branch first, and then do a merge into `main` 
+     after-wards.
```

Comment:
Note I think we want to avoid things like bug fixes ending up on release branches _first_, because:

- generally the set of CI gates is best and most maintained on main, so a change that lands there first will have passed all these.
- there's the real risk of post-completion errors -- ie. we make the panic fix-the-bug release, but then forget to get the fix and its regression test onto main. keeping always the latest branch avoids that.

Reply:
Those are convincing reasons to me. I can back out this part of the recommendations.

> generally the set of CI gates is best and most maintained on main

As a concrete example it looks like `rel-0.21` had been missing the cross compile check :sweat:

Reply:
Dropped the note about preferring merge to feature branch first. Also added a note when creating a feature branch to create a branch protection rule for it matching the settings from `main`.